services/basic_types.proto

```diff
@@ -79,9 +79,16 @@ message AccountID {
     int64 realmNum = 2;
 
     /**
-     * A nonnegative account number unique within its realm
-     */
-    int64 accountNum = 3;
+     * An account number unique within its realm which can be either a non-negative integer or an alias public key.
+     * ThresholdKey and KeyList are not supported for alias and only a primitive key type is supported.
+     * Alias public key is used for auto account creation using a crypto transfer. Once an alias key is set for an
+     * account it is immutable.
+     */
+    oneof account {
+        int64 accountNum = 3;
```

Comment:
As a style consideration, can each field in the protobuf get a documentation comment rather than the oneof container?  

I don't see anywhere else where the documentation is on the oneof in lieu of being on the fields themselves, and in all such cases each field in the oneof has documentation. In the single case where oneof has a doc it is because we changed the underlying type of a data field from a protobuf string to a StringValue.  The fields both had the same contextual meaning.

Reply:
Thanks @shemnon . Updated documentation for each field.